### PR TITLE
Rename _legacy to _adjust_to_cased

### LINKED ISSUE #3801: Decide on naming of "legacy" titlecase adjustment mode
Regarding `new_legacy()` on `TitlecaseMapper`:

> Thought: I wonder if this should be called `new_spec()` or `new_chapter3()` or similar, to take a positive angle toward what this is, rather than the negative term "legacy" which is not very specific.

_Originally posted by @sffc in https://github.com/unicode-org/icu4x/pull/3779#discussion_r1284049394_

cc @markusicu @sffc

### PR DESCRIPTION
**Pull request by Manishearth**
Uses @markusicu's suggestion from #3801 (fixes #3801).

Seems fine, though `adjust_to_cased_to_string()` is kinda clunky.

This is also the final open stabilization issue, so I'm going to mark this as fixing https://github.com/unicode-org/icu4x/issues/3234 . (note for reviewers: if you think there are still outstanding items, feel free to edit the magic issue-closing comment out of this PR body)

### CODE REVIEW

File: components/casemap/src/casemapper.rs [hunk 1]
```diff
@@ -146,16 +146,16 @@ impl CaseMapper {
     /// as a `LanguageIdentifier` (usually the `id` field of the `Locale`) if available, or
     /// `Default::default()` for the root locale.
     ///
-    /// This function performs legacy head adjustment behavior when [`HeadAdjustment::Adjust`] is set. See
+    /// This function performs "adjust to cased" head adjustment behavior when [`HeadAdjustment::Adjust`] is set. See
```
Comment (markusicu) [inline components/casemap/src/casemapper.rs:149]: "head adjustment" is confusing. I have never heard "head" used in discussion of titlecasing.
How about "titlecasing index adjustment" or similar?

Comment (Manishearth) [reply components/casemap/src/casemapper.rs:149]: We've been calling this HeadAdjustment and TailCasing which I think has a nice symmetry. "Index adjustment" feels more internal, I think "head" is quite clear personally.

Comment (sffc) [reply components/casemap/src/casemapper.rs:149]: Bikeshed:

- Head/Tail
- Lead/Trail
- Start/Remainder

File: components/casemap/src/titlecase.rs [hunk 2]
```diff
@@ -159,14 +159,14 @@ impl TitlecaseMapper<CaseMapper> {
...
-    /// A constructor which creates a [`TitlecaseMapper`] using compiled data, with the legacy head adjustment behavior.
+    /// A constructor which creates a [`TitlecaseMapper`] using compiled data, with the "adjust to cased" head adjustment behavior.
     /// See struct docs on [`TitlecaseMapper`] for more information on head adjustment behavior and usage examples.
     ///
     /// ✨ *Enabled with the `compiled_data` Cargo feature.*
     ///
     /// [📚 Help choosing a constructor](icu_provider::constructors)
     #[cfg(feature = "compiled_data")]
-    pub const fn new_legacy() -> Self {
+    pub const fn new_adjust_to_cased() -> Self {
```
Comment (sffc) [inline components/casemap/src/titlecase.rs:169]: Suggestion: on second thought, we don't really need this constructor. People can use `CaseMapper::titlecase_segment_adjust_to_cased` if they don't want to load the extra data.

Comment (Manishearth) [reply components/casemap/src/titlecase.rs:169]: What if they want to still have it be the same type, though? Might make it clunky

Comment (sffc) [reply components/casemap/src/titlecase.rs:169]: It's kind-of confusing that `HeadAdjustment` has 2 choices but there are 3 behaviors. We can add all 3 behaviors to that enum. Over in `CaseMapper::titlecase_segment_adjust_to_cased`, we decide what to do if given the enum variant which we don't support.

Comment (Manishearth) [reply components/casemap/src/titlecase.rs:169]: yeah but the choice of behaviors affects data; we had kinda discussed this model and the 2x2 chart with the merged cells

Comment (sffc) [reply components/casemap/src/titlecase.rs:169]: People who care about their data have a path; people who don't care as much about their data have a cleaner API. I think that's slightly better than a confusing API that affects everyone.

Comment (Manishearth) [reply components/casemap/src/titlecase.rs:169]: Are you proposing we remove _specifically_ the compiled_data adjust_to_cased ctor? That seems okay.

Comment (Manishearth) [reply components/casemap/src/titlecase.rs:169]: What should we do over FFI, then?

**Comment by Manishearth** [comment]
Superseded by https://github.com/unicode-org/icu4x/pull/3843

**Pull request closed without merging**